lib/simple_navigation/core/item.rb

```diff
@@ -145,7 +152,7 @@ def selected_by_highlights_on?
       when Regexp then SimpleNavigation.request_uri =~ highlights_on
       when Proc then highlights_on.call
       when :subpath
-        escaped_url = Regexp.escape(url_without_anchor)
+        escaped_url = Regexp.escape(clean_url)
         !!(SimpleNavigation.request_uri =~ /^#{escaped_url}(\/|$|\?)/i)
```

Comment:
Not sure if this is really necessary. If you look at the regexp on this line, it matches any URL containing ? Can you give an example of an URL where autohighlights fails?